### ISSUE TITLE
Replace all uses of % string formatting by str.format

### ISSUE THREAD

**Issue by duk3luk3** [FuelRats/pipsqueak#20]
`%` is deprecated, please stop using it. `str.format` may be a bit clunkier and more verbose in some cases but it's a much better interface.

**Comment by tyrope** [comment]
Shouldn't be too hard, just so that I'm correct, an example:

```
'Error fetching data: [%s]%s' % (ans['code'], ans['details'])
```

becomes

```
'Error fetching data: [{0}]{1}'.format(ans['code'], ans['details'])
```

**Comment by duk3luk3** [comment]
Yes, or even

`'Error fetching data: [{0:code}]{0:details}'.format(ans)`

:-)

**Comment by tyrope** [comment]
For some reason GitHub didn't do it's magic. Fix PR waiting, number #22

**Comment by tyrope** [comment]
> Yes, or even
> 
> `'Error fetching data: [{0:code}]{0:details}'.format(ans)`

```
Python 2.7.4 (default, Apr  6 2013, 19:55:15) [MSC v.1500 64 bit (AMD64)] on win32
>>> d = {'foo': 'value1', 'bar': 'value2'}
>>> 'The dict has 2 values: {0:foo} and {0:bar}'.format(d)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: Invalid conversion specification
```

This _might_ affect the above mentioned PR.
workaround:

```
Python 2.7.4 (default, Apr  6 2013, 19:55:15) [MSC v.1500 64 bit (AMD64)] on win32
>>> d = {'foo': 'value1', 'bar': 'value2'}
>>> 'The dict has 2 values: {0} and {1}'.format(d['foo'],d['bar'])
'The dict has 2 values: value1 and value2'
```

edit Dec 14th:
Proper fix:

```
Python 2.7.4 (default, Apr  6 2013, 19:55:15) [MSC v.1500 64 bit (AMD64)] on win32
>>> d = {'foo': 'value1', 'bar': 'value2'}
>>> 'The dict has 2 values: {0[foo]} and {0[bar]}'.format(d)
'The dict has 2 values: value1 and value2'
```
